apiserver/facades/agent/machine/machiner.go

```diff
@@ -109,6 +123,7 @@ func NewMachinerAPIForState(
 	modelInfoService ModelInfoService,
 	networkService NetworkService,
 	machineService MachineService,
+	applicationService ApplicationService,
```

Comment:
Here's the reason for my previous example: we're injecting a service that is only needed for the life getter, if it took a domain services then there'd be no need for this.